### DISCUSSION TITLE
FCST_SEGMENTS error

### DISCUSSION THREAD

**Discussion by benjamin-cash** [NOAA-EMC/global-workflow#2854] [Q&A] [answered]
I have been getting an error message from `setup_xml.py` when I run the global workflow on Frontera. The message is:
```
File "/work2/02441/bcash/frontera/SFS_baseline/workflow/applications/applications.py", line 81, in __init__
    raise ValueError(f'Forecast segments do not increase monotonically: {",".join(self.fcst_segments)}')
ValueError: Forecast segments do not increase monotonically: 0,,,4,8,,,1,2,0
```

Looking at where this is set, I find in `config.base`:
`export FCST_SEGMENTS="${FHMIN_GFS},${breakpnts:+${breakpnts},}${FHMAX_GFS}"`

Digging further, `FHMIN_GFS` is set to 0, `FHMAX_GFS` is set to 120, and `breakpnts` is set to 48. So what should be `0,48,120` is getting mangled somehow by `applications.py` and causing an error. 

Interestingly, this setting doesn't appear at all in the experiment I set up on Hercules, which makes me think there is some kind of machine-specific setting that is causing it. Any help in tracking this down and getting past it would be greatly appreciated!

**Comment by WalterKolczynski-NOAA** [comment, marked as answer]
Segments were only [added last week](https://github.com/NOAA-EMC/global-workflow/pull/2795), so if your Hercules experiment is from before then, it wouldn't have the setting. That PR also updated the gfs_utils version, so make sure you have updated your submodules if this wasn't a new clone.

**Reply by benjamin-cash** [reply]
I have been running using @NeilBarton-NOAA 's SFS_baseline branch in his fork, which doesn't include that PR. I must have somehow jumped branches on Frontera and pulled in the newer code without realizing it. Thanks for clearing that up and the quick reply!

**Comment by benjamin-cash** [comment]
Hi @WalterKolczynski-NOAA - I am back to trying to set up a run on Hercules, and I am still getting the following error:
```
[cash@hercules-login-3 workflow]$ ./setup_xml.py /work/noaa/nems/cash/UFS-coupled/C96_19940501
Traceback (most recent call last):
  File "/work/noaa/nems/cash/global-workflow/workflow/./setup_xml.py", line 79, in <module>
    main()
  File "/work/noaa/nems/cash/global-workflow/workflow/./setup_xml.py", line 70, in main
    app_config = app_config_factory.create(f'{net}_{mode}', cfg)
  File "/home/cash/.local/lib/python3.10/site-packages/wxflow/factory.py", line 71, in create
    return self._builders[key](*args, **kwargs)
  File "/work/noaa/nems/cash/global-workflow/workflow/applications/applications.py", line 21, in __call__
    obj = type.__call__(cls, *args, **kwargs)
  File "/work/noaa/nems/cash/global-workflow/workflow/applications/gefs.py", line 11, in __init__
    super().__init__(conf)
  File "/work/noaa/nems/cash/global-workflow/workflow/applications/applications.py", line 73, in __init__
    raise ValueError(f'Forecast segments do not increase monotonically: {",".join(self.fcst_segments)}')
ValueError: Forecast segments do not increase monotonically: 0,,,4,8,,,1,2,0
```
My experiment setup command was:
`[cash@hercules-login-3 workflow]$ ./setup_expt.py gefs forecast-only --idate 19940501 --edate 19940502 --app S2S --start cold --gfs_cyc 1 --resdetatmos 96 --resensatmos 96 --resensatmos 96 --resdetocean 1 --comroot /work/noaa/nems/cash/UFS-coupled/comrot --expdir /work/noaa/nems/cash/UFS-coupled --nens 10 --pslot C96_19940501`
